src/Plugin/tmgmt/Translator/PermissionTranslator.php

```diff
@@ -0,0 +1,534 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_translation\Plugin\tmgmt\Translator;
+
+use Drupal\tmgmt_content\Plugin\tmgmt\Source\ContentEntitySource;
+use Drupal\node\NodeInterface;
+use Drupal\Core\Access\AccessManagerInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Url;
+use Drupal\oe_translation\AlterableTranslatorInterface;
+use Drupal\oe_translation\LocalTranslatorInterface;
+use Drupal\oe_translation\RouteProvidingTranslatorInterface;
+use Drupal\tmgmt\ContinuousTranslatorInterface;
+use Drupal\tmgmt\JobInterface;
+use Drupal\tmgmt\JobItemInterface;
+use Drupal\tmgmt\TranslatorInterface;
+use Drupal\tmgmt\TranslatorPluginBase;
+use Drupal\tmgmt_local\LocalTaskInterface;
+use Drupal\tmgmt_local\LocalTaskItemInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Drupal current user provider.
+ *
+ * @TranslatorPlugin(
+ *   id = "permission",
+ *   label = @Translation("Permission"),
+ *   description = @Translation("Allows the users with a certain permission to translate content."),
+ *   ui = "\Drupal\oe_translation\PermissionTranslatorUI",
+ *   default_settings = {
+ *     "permission" = "translate any entity"
+ *   },
+ *   map_remote_languages = FALSE
+ * )
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
```

Comment:
I don't think we should supress this since it indicates what might be a deeper problem with this class: we have added so many interfaces and extra functionality to this class that its no longer clear what its purpose is. Its supposed to be a translator, essentially taking care of formatting and requesting a translation, but it also alters breadcrumbs, checks for access on items and so on. I think we need to take a deeper look at this and maybe come up with a different solution.

Reply:
@imanoleguskiza We changed the architecture to allow translators to do more things. Thats because we need them to do more things. For example, the PermissionTranslator needs to ensure that the user has access to translate if they have a permission! Hence, we came up with this approach to delegate everything specific to that translator. The Poetry translatoir wont have that cause it wont implement that interface. It will have other things related to the business requirements.